Assignment7-(optional)/question1.js

```diff
@@ -0,0 +1,139 @@
+/*
+    Advanced Assignments: (optional)
+    ● Write a function to check deep equality of two nested objects/arrays
+    ● Write a recursive function to create a deep clone of a nested object
+    ● Write a function that returns a nested key within an object:
+    Example:
+
+    {
+        name:”Harry Potter”,
+        age: 12,
+        address: {
+            details: [“4”, “Privet Drive”],
+            area:”Little Whinging”,
+            city: “Surrey”,
+            state: “England”
+        } 
+    }
+    getNestedKey(obj, “address.details.1”)
+    This should return “Privet Drive”
+*/
+
+function checkDeepCopyObject(obj1, obj2) {
```

Comment:
pass all cases of non-equality to this function and also print output why it is not equal before returning from that particular case.

Reply:
done